sql_metadb/derived_tables/users_departments_unpacked.sql

```diff
@@ -0,0 +1,52 @@
+DROP TABLE IF EXISTS users_departments_unpacked;
+
+-- Create a derived table that takes the users table and joins
+-- in the department information
+CREATE TABLE users_departments_unpacked AS
+WITH departments_array AS (
+    SELECT
+        uu.id::uuid AS user_id,
+        departments.jsonb #>> '{}'AS department_id,
+        departments.ordinality AS department_ordinality
+    FROM
+        folio_users.users AS uu
+        CROSS JOIN LATERAL jsonb_array_elements(jsonb_extract_path(jsonb, 'departments'))
+        WITH ORDINALITY AS departments (jsonb)
+)
+SELECT
+    departments_array.user_id,
+    departments_array.department_id::uuid,
+    departments_array.department_ordinality,
+    jsonb_extract_path_text(jsonb, 'name') AS department_name,
+    jsonb_extract_path_text(jsonb, 'code') AS department_code
+    --jsonb_extract_path_text(jsonb, 'usageNumber') AS department_usage_number
+FROM
+    departments_array
+LEFT JOIN folio_users.departments AS ud
```

Comment:
This works, however, if you use `folio_users.departments__t` you don't have to do the JSON extract. It can be done either way :)